app/domains/ranks/insanity_scores.rb

```diff
@@ -12,7 +12,9 @@ def ranking_criterion(score)
       completions = @completions.fetch(score[:user_id], [])
       [
         score[:score],
-        -completions.sum(&:duration)
+        completions.count,
+        -completions.sum { |completion| completion.duration || 0 },
```

Comment:
It the tests it seems like you want to sort by putting the fastest players first but here you're sorting them by -speed which means the slowest will be first
I changed this and your test passed
```suggestion
          completions.sum { |completion| completion.duration || 0 },
```